File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int n = nums.length;
        int totalSum = 0;

        for(int i = 0; i < n;i++){
            totalSum += nums[i];
        }

        int leftSum = 0;
        int rightSum = totalSum;

        for(int j = 0; j < n;j++) {
            rightSum = rightSum - nums[j];
            //check before manipulating leftSum
            if(leftSum == rightSum) return j;
            leftSum = leftSum + nums[j];
        }

        return -1;
    }
}

File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        n = len(nums)

        totalSum = sum(nums)

        leftSum = 0
        rightSum =totalSum

        for i in range(n):
            rightSum -= nums[i]
            # check if before manipulating the left Sum
            if leftSum == rightSum:
                return i
            leftSum += nums[i]
        
        return -1

        